tests/platform_tests/mellanox/util.py

```diff
@@ -0,0 +1,329 @@
+                                                        "TxPowerLowAlarm": pattern_power,
+                                                        "TxPowerLowWarning": pattern_power}
+            expected_threshold_data_keys_and_pattern.update(expected_channel_threshold_values_keys_and_value_pattern)
+            expected_threshold_data_keys_and_pattern.update(expected_module_threshold_values_keys_and_value_pattern)
+
+            logging.info("Check if MonitorData's keys exist and the corresponding value format is correct")
+            check_dom_monitor_key_and_data_format(expected_monitor_data_keys_and_pattern,
+                                                  sfp_eeprom_info["MonitorData"])
+
+            logging.info("check if ThresholdData's keys exist and the corresponding value format is correct")
+            check_dom_monitor_key_and_data_format(expected_threshold_data_keys_and_pattern,
+                                                  sfp_eeprom_info["ThresholdData"])
+
+        else:
+            if "ChannelThresholdValues" not in excluded_keys:
+                logging.info(
+                    "Check if ChannelThresholdValues' keys exist and the corresponding value format is correct")
+                check_dom_monitor_key_and_data_format(expected_channel_threshold_values_keys_and_value_pattern,
+                                                      sfp_eeprom_info["ChannelThresholdValues"])
+
+            if "ModuleThresholdValues" not in excluded_keys:
+                logging.info("Check if ModuleThresholdValues' keys exist and the corresponding format is correct")
+                check_dom_monitor_key_and_data_format(expected_module_threshold_values_keys_and_value_pattern,
+                                                      sfp_eeprom_info["ModuleThresholdValues"])
+
+            logging.info("Check if ChannelMonitorValues's value format is correct")
+            for k, v in sfp_eeprom_info["ChannelMonitorValues"].items():
+                pattern = pattern_power if "Power" in k else pattern_bias
+                assert re.match(pattern, v), "Value of {}:{} format is not correct. pattern is {}".format(k, v, pattern)
+
+            logging.info("Check ModuleMonitorValues keys exist and the corresponding value format is correct")
+            check_dom_monitor_key_and_data_format(expected_module_monitor_values_keys_and_pattern,
+                                                  sfp_eeprom_info["ModuleMonitorValues"])
+
+
+def check_dom_monitor_key_and_data_format(expected_keys_and_pattern_dict, dom_monitor_data):
+    for key, pattern in expected_keys_and_pattern_dict.items():
+        assert key in dom_monitor_data, "Key {} doesn't exist in {}".format(key, dom_monitor_data)
+        assert re.match(pattern, dom_monitor_data[key]), \
+            "Value of {}:{} format is not correct. pattern is {}".format(
+                key, dom_monitor_data[key], pattern)
+
+
+def is_support_dom(duthost, port_index):
```

Comment:
Maybe it is better to use `ethtool -m sfp{index}`. Part of the data of "show interface transceiver eeprom -d" comes from `sfp.get_transceiver_bulk_status`. So, using `sfp.get_transceiver_bulk_status` to test "show interface transceiver eeprom -d" may not find real issues.

Reply:
a specific field of the EEPROM can be different in the output of the "ethtool -m" and SONiC CLI output, no easy way to verify the SONIC CLI output with it.

Reply:
maybe can use mlxlink/ethtool to check whether relevant fields are valid.